### PR TITLE
refactor: Use new filtering params for Layman requests in compositions/add-data

### PR DESCRIPTION
**Pull request by DailisLangovskis**

### CODE REVIEW

File: projects/hslayers/src/components/compositions/endpoints/compositions-layman.service.ts [hunk 1]
```diff
@@ -70,14 +118,18 @@ export class HsCompositionsLaymanService {
       );
     return endpoint.listLoading;
   }
-  compositionsReceived(endpoint: HsEndpoint, params, response): void {
-    if (!response && response.length == 0) {
+  compositionsReceived(endpoint: HsEndpoint, response): void {
+    if (!response.body && response.body.length == 0) {
```
Comment (raitisbe) [inline projects/hslayers/src/components/compositions/endpoints/compositions-layman.service.ts:122]: Can both conditions even be true at the same time?

**Comment by FilipLeitner** [comment]
Have seen it already and discussed with Dailis, theres a fix needed because Layman layers are not visible when onlyMine filter is false

**Comment by FilipLeitner** [comment]
Keep thinking about the "filter by map extent"  and "sort by BBOX" interaction. Dont think that user will understand  what it means or know what to do when a error message " Value "bbox" of parameter "order_by" can be used only if "bbox_filter" or "bbox" parameter is set." pop up. 
We could either disable this combination  or at least we should make the message more explanatory from the user point of view.
What do you think?

**Comment by DailisLangovskis** [comment]
> Keep thinking about the "filter by map extent" and "sort by BBOX" interaction. Dont think that user will understand what it means or know what to do when a error message " Value "bbox" of parameter "order_by" can be used only if "bbox_filter" or "bbox" parameter is set." pop up.
> We could either disable this combination or at least we should make the message more explanatory from the user point of view.
> What do you think?

Hmm.. mby we should create a dialog to inform the user and ask if user wants to enable filter by extent to continue.

File: projects/hslayers/src/components/add-data/catalogue/micka/micka.service.ts [hunk 2]
```diff
@@ -109,6 +109,12 @@ export class HsMickaBrowserService {
       'EPSG:4326'
     );
     const bbox = data.filterByExtent ? "BBOX='" + b.join(' ') + "'" : '';
+    const sortBy =
+      query.sortby !== undefined && query.sortby != ''
+        ? query.sortby == 'date'
+          ? 'date:D'
```
[the remark this answers is not in the document]
Comment (DailisLangovskis) [reply projects/hslayers/src/components/add-data/catalogue/micka/micka.service.ts:115]: No I actually did this on purpose.

File: projects/hslayers/src/assets/locales/cs.json [hunk 3]
```diff
@@ -202,7 +202,11 @@
     "contactInformation": "Kontaktní informace",
     "warning": "Varování",
     "noDataReceived": "Nebyla přijata žádná data!",
-    "or": "anebo"
+    "or": "anebo",
+    "wrongCombinationOfParams": "Neplatná kombinace parametru",
+    "mapExtentFilterMissing": "Řazení výsledku dle rozsahu dat je možné jenom po aktivaci 'filtrace na rozsah okna'. Aktivujte jej prosím.",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:207]:
```suggestion
    "mapExtentFilterMissing": "Řazení výsledků dle rozsahu dat je možné jen po aktivaci 'filtrace na rozsah okna'. Aktivujte ji prosím.",
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 4]
```diff
@@ -202,7 +202,11 @@
...
+    "unknownError": "Neznáma chyba. Zkuste to znova, prosím.",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:208]:
```suggestion
    "unknownError": "Neznámá chyba. Zkuste to znovu, prosím.",
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 5]
```diff
@@ -202,7 +202,11 @@
...
+    "Unsuccessful OAuth2 authentication. Access token is not valid": "Neuspešná autentizace. Neplatný přístupový token"
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:209]:
```suggestion
    "Unsuccessful OAuth2 authentication. Access token is not valid": "Neúspěšná autentizace OAuth2. Neplatný přístupový token"
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 6]
```diff
@@ -651,5 +655,14 @@
     "totalDistance": "Celková vzdálenost",
     "waypointDrawingHint": "Klikněte prosím do mapy pro přidání bodů trasy",
     "waypoints": "Body trasy"
+  },
+  "ERRORMESSAGES": {
+    "0": "Chybná odpověď pro {{url}}: 0 Neznáma chyba",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:660]:
```suggestion
    "0": "Chybná odpověď pro {{url}}: 0 Neznámá chyba",
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 7]
```diff
@@ -651,5 +655,14 @@
...
+    "400": "Chybná odpověď pro {{url}}: 400 Nesprávny požadavek",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:661]:
```suggestion
    "400": "Chybná odpověď pro {{url}}: 400 Špatný požadavek",
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 8]
```diff
@@ -651,5 +655,14 @@
...
+    "403": "Chybná odpověď pro {{url}}: 403 Odmítnuto",
+    "404": "Chybná odpověď pro {{url}}: 404 Nenalezeno",
+    "502": "Chybná odpověď pro {{url}}: 502 Nesprávna brána",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:664]:
```suggestion
    "502": "Chybná odpověď pro {{url}}: 502 Špatná brána",
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 9]
```diff
@@ -651,5 +655,14 @@
...
+    "Timeout has occurred": "Chybná odpověď pro {{url}}: Čas vypršel",
+    "Invalid Params": "Před sdílením mapy se ujistete zdali jsou všechna vyžadována pole správne vyplnená!"
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:666]:
```suggestion
    "Invalid Params": "Před sdílením mapy se ujistěte zda jsou všechna vyžadovaná pole správně vyplněná!"
```

File: projects/hslayers/src/assets/locales/en.json [hunk 10]
```diff
@@ -202,7 +202,12 @@
     "contactInformation": "Contact Information",
     "noDataReceived": "No data received!",
     "warning": "Warning",
-    "or": "or"
+    "or": "or",
+    "wrongCombinationOfParams": "Wrong combination of parameters",
+    "mapExtentFilterMissing": "Please enable Filter by map extent checkbox to by able to sort data by Bounding box",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/en.json:207]:
```suggestion
    "mapExtentFilterMissing": "Please enable Filter by map extent checkbox to be able to sort data by Bounding box",
```

**Review by jmacura — CHANGES_REQUESTED** [review]
I propose some polishing of translations (grammar nazi 👮 😳 )

**Comment by FilipLeitner** [comment]
> I propose some polishing of translations (grammar nazi 👮 😳 )

Great, thanks! Struggled a with some of them so I hoped something like this would come

**Review by raitisbe — APPROVED** [review]

**Pull request merged**